src/System/PosixCompat/Process.hs

```diff
@@ -0,0 +1,24 @@
+{-# LANGUAGE CPP #-}
+
+{-|
+This module makes the operations exported by @System.Posix.Process@
+available on all platforms. On POSIX systems it re-exports operations from
+@System.Posix.Process@, on other platforms it emulates the operations as far
+as possible.
+-}
+module System.PosixCompat.Process (
+      getProcessID
+    ) where
+
+#ifndef mingw32_HOST_OS
+
+import System.Posix.Process
+
+#else
+
+import System.Posix.Types (ProcessID)
+
+getProcessID :: IO ProcessID
+getProcessID = fromIntegral <$> System.Win32.Process.getCurrentProcessID
```

Comment:
CI fails here on Windows: https://github.com/haskell-pkg-janitors/unix-compat/actions/runs/6729599930/job/18346250074?pr=6#step:5:122
``` 
src\System\PosixCompat\Process.hs:22:33: error:
Error:     Not in scope: ‘System.Win32.Process.getCurrentProcessID’
    No module named ‘System.Win32.Process’ is imported.
   |
22 | getProcessID = fromIntegral <$> System.Win32.Process.getCurrentProcessID
```

Reply:
Whoops! Hopefully this fixes it.